src/nvim/edit.c

```diff
@@ -3419,6 +3419,7 @@ static void ins_compl_fixRedoBufForLeader(char_u *ptr_arg)
     else
       return;        /* nothing to do */
   }
+  assert(ptr != NULL);
```

Comment:
Seems very weird that PVS studio thinks `ptr` can be null here. Could the checks above be optimized out somehow?